Add clientset for Output CRD

mugdha-adhav opened the pull request:
## Description
- Added client code for Output CRD using code-generator
- Clientset is generated for non-namespace scoped resource
- `client` packge is auto-generated using code-generator
## Purpose
- The newly added clientset will enable us to create custom-resources of type Output
## Testing Done
- Manually tested for kubernetes 1.15.2
- Made sure that the custom resources of type **Output** are created using the clientset

mugdha-adhav committed:
Add clientset for Output CRD

texascloud commented:
What does this change do?

smanpathak commented:
> What does this change do?

@CamelCaseNotation I am going to add a template for pull request. Devs forget such things when posting reviews!

smanpathak commented on pkg/client/clientset/versioned/clientset.go:
```diff
@@ -0,0 +1,98 @@
+/*
+Copyright The Kubernetes Authors.
```
Lets change the copyright Platform9 Systems

mugdha-adhav replied:
Made the requested changes

smanpathak commented on pkg/client/clientset/versioned/clientset.go:
```diff
@@ -0,0 +1,98 @@
+	flowcontrol "k8s.io/client-go/util/flowcontrol"
+)
+
+type Interface interface {
```
Put comment describing the interface. Also, don't call this "interface"

mugdha-adhav replied:
Renamed Interface to **LoggingV1alpha1Interface** and added comment

smanpathak commented on pkg/client/clientset/versioned/clientset.go:
```diff
@@ -0,0 +1,98 @@
+	return c.loggingV1alpha1
+}
+
+// Deprecated: Logging retrieves the default version of LoggingClient.
```
why is this depreciated? It's strange to see such language in new code we are adding.

mugdha-adhav replied:
Removed deprecated implementations. The implementations are anyways not needed for us.

smanpathak commented on pkg/client/clientset/versioned/doc.go:
```diff
@@ -0,0 +1,20 @@
+/*
+Copyright The Kubernetes Authors.
```
change to Platform9. See other files for reference.

mugdha-adhav replied:
Made the requested changes

smanpathak commented:
@mugdha-adhav Overall, please add unit tests.

mugdha-adhav committed:
Modified Output clientset

- Changed copyright for logging clientset to platform9
- Modified clientset for non-namespaced resources
- Removed deprecated implementations
- Renamed logging interface to LoggingV1alpha1Interface and added comment
- Fixed the version to v1alpha1 in the Scheme

mugdha-adhav commented:
> What does this change do?

@CamelCaseNotation @smanpathak added the description for the PR

mugdha-adhav committed:
Add unit tests for Output Custom Resource in clientset

smanpathak reviewed (approved):
Ok to submit, after fixing the failing unit test

mugdha-adhav committed:
Restructured and fixed unit test for output clientset

Added subtests for each test
Add check for result type
Add check for output object
